### PR TITLE
add a file storage engine

### PR DESCRIPTION
**Pull request by djekl**

### CODE REVIEW

File: src/Moltin/Cart/Storage/LaravelFile.php [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+{
+    protected $storagePath;
+
+    protected $identifier;
+
+    protected static $cart = array();
+
+    public function __construct()
+    {
+        $this->storagePath = storage_path('Cart');
```
Comment (outrunthewolf) [inline src/Moltin/Cart/Storage/LaravelFile.php]: I know this is an Laravel4 function, is it available in L5? And, Is it worth making the name 'Cart' configurable in some way?

File: src/Moltin/Cart/Storage/LaravelFile.php [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+     * Retrieve the cart data
+     *
+     * @param bool $asArray
+     * @return array
+     */
+    public function &data($asArray = false)
+    {
+        $cart =& static::$cart[$this->id];
+
+        if (! $asArray) {
```
Comment (outrunthewolf) [inline src/Moltin/Cart/Storage/LaravelFile.php]: Space on !

can we standardise throughout

File: src/Moltin/Cart/Storage/LaravelFile.php [hunk 3]
```diff
@@ -0,0 +1,203 @@
+<?php
+
+/**
+ * This file is part of Moltin Cart, a PHP package to handle
+ * your shopping basket.
+ *
+ * @package moltin/cart
+ * @author Alan Wynn <hello@alanwynn.me>
+ * @link http://github.com/moltin/cart
```
Comment (outrunthewolf) [inline src/Moltin/Cart/Storage/LaravelFile.php]: Correct URL for package

File: src/Moltin/Cart/Storage/LaravelFile.php [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+     * Return the current cart identifier
+     *
+     * @return void
+     */
+    public function getIdentifier()
+    {
+        return $this->identifier;
+    }
+
+    protected function saveCart()
```
Comment (outrunthewolf) [inline src/Moltin/Cart/Storage/LaravelFile.php]: Doc block on function.

I know someone hasn't bothered with the other classes, but we may as well make it nice

**Comment by djekl** [comment]
I have fixed the things you pointed out

**Comment by djekl** [comment]
Oh, and I forgot to say, `storage_path()` is a L5 helper function - http://laravel.com/docs/5.1/helpers#paths

**Comment by outrunthewolf** [comment]
Awesome :+1:

**Pull request merged**